remoting/src/main/java/org/apache/rocketmq/remoting/common/RemotingUtil.java

```diff
@@ -145,8 +144,10 @@ public static String normalizeHostAddress(final InetAddress localHost) {
     }
 
     public static SocketAddress string2SocketAddress(final String addr) {
-        String[] s = addr.split(":");
-        InetSocketAddress isa = new InetSocketAddress(s[0], Integer.parseInt(s[1]));
+        int split = addr.lastIndexOf(":");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the lastIndexOf() search on line 147 does not find the substring and returns -1, the call to substring() on line 148 will throw IndexOutOfBoundsException.